File: 2023/May/week1/정환훈/2_택배 배달과 수거하기.cpp
#include <string>
#include <vector>

using namespace std;

long long solution(int cap, int n, vector<int> del, vector<int> pick) {
    long long answer = 0;
    int d = 0;
    int p = 0;    
    //   for i in range(n-1, -1, -1):
    for(int i=n-1;i>=0;i--){
        long long cnt = 0;
        d -= del[i];
        p -= pick[i];
        while(d < 0 || p < 0){
            d += cap;
            p += cap;
            cnt++;
        }
        answer += (i+1)*2*cnt;
    }    
    return answer;    
}

File: 2023/May/week1/이정욱/2_택배 배달과 수거하기.py
def solution(cap, n, deliveries, pickups):
    answer = 0
    
    deliveries = deliveries[::-1]
    pickups = pickups[::-1]
    
    d = 0
    p = 0
    
    for i in range(n):
        d += deliveries[i]
        p += pickups[i]
        
        while d > 0 or p > 0:
            d -= cap
            p -= cap
            answer += (n-i)*2
    return answer